gammapy/maps/core.py

```diff
@@ -1095,6 +1097,57 @@ def reproject_to_geom(self, geom, preserve_counts=False, precision_factor=10):
             output_map = output_map.resample(geom, preserve_counts=preserve_counts)
         return output_map
 
+    def reproject_by_image(
+        self,
+        geom,
+        preserve_counts=False,
+        precision_factor=10,
+    ):
+        """Reproject each image of a ND map to input 2d geometry.
+
+           For large maps this method is faster than `reproject_to_geom`.
```

Comment:
```suggestion
       For large maps this method is faster than `reproject_to_geom`.
```

Reply:
Maybe mention in the docstring that this can be sued with mutli-processing context manager ?